File: server/python/controllers/client_token.py
import os
from flask import jsonify, make_response
from .gateway import braintree_gateway

DOMAINS = os.getenv("DOMAINS", "").split(",")


def braintree_sdk_token():
    try:
        client_token = braintree_gateway().client_token.generate({"domains": DOMAINS})
        return jsonify(clientToken=client_token)
    except Exception as error:
        print(error)
        return make_response(jsonify(error=str(error)), 500)

File: server/node/src/controllers/client-token.js
import { getGateway } from '../lib/gateway.js';

const { DOMAINS } = process.env;

export function getClientToken(_req, res) {
  try {
    getGateway().clientToken.generate(
      {
        domains: DOMAINS.split(','),
      },
      (error, response) => {
        if (error) {
          console.error(error);
          res.status(500).json({ error: error.message });
          return;
        }
        res.json({ clientToken: response.clientToken });
      },
    );
  } catch (error) {
    console.error(error);
    res.status(500).json({ error: error.message });
  }
}
